File: java/Bill_due.java
import java.util.Scanner;

public class Bill_due{
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n,k,sum=0,paid;
        n=in.nextInt();
        k=in.nextInt();
        int[] orders = new int[n];
        for(int i =0;i<n;i++){
            orders[i]=in.nextInt();
            sum+=orders[i];
        }
        paid=in.nextInt();
        if(paid==((sum-orders[k])/2))
            System.out.println("Bon Appetit");
        else
        System.out.println(paid-(sum-orders[k])/2);
        
    }
}

File: C/Bill_Due.c
#include <stdio.h>
int main(){
    int n,k,sum=0,paid,tmp;
    scanf("%d",&n);
    scanf("%d",&k);
    for(int i =0;i<n;i++){
        scanf("%d",&tmp);
        if(k!=i){
        sum+=tmp;}
    }
    scanf("%d",&paid);
    if(paid==sum/2){
        printf("Bon Appetit");}
    else{
    printf("%d",paid-sum/2);}
    return 0;
}